pretext/ExtraTopics/BigO.ptx

```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0"?>
+<section xml:id="extratopics-bigo">
+  <title>Big O Analysis</title>
+
+  <p>It's impossible to experimentally measure the theoretical performance of algorithms on a computer as the size of our data (which we call n, to some factor) goes to infinity,
+    or gets really big. We won't get deep into it, but this is due to things like variable hardware between computers and just general performance of software.</p>
+
+  <p>This is where <term>Big O Analysis</term> comes in. We can measure the theoretical performance of algorithms, or their <term>complexity</term>,
+    in a special mathematical notation.</p>
+
+  <p>This mathematical notation is denoted with a capital O (a <em>big o</em>!) followed by parentheses.
+    Inside of the <c>O()</c> is most commonly a <c>1</c> or a <c>n</c> raised to some mathematical factor.</p>
+
+  <p>These factors have <em>magnitude</em>, which helps us to quantify how quick or slow an algorithm is. 
+    From left to right, left being the quickest time and right being the slowest time, we typically see these complexities:</p>
+
+  <p><c>O(1)</c>, <c>O(logn)</c>, <c>O(n)</c>, <c>O(nlogn)</c>, <c>O(n^2)</c>, <c>O(n^3)</c>, <c>O(2^n)</c>, <c>O(n!)</c>.</p>
+
```

Comment:
Is this already in the slides? I can't remember. This would be a good thing to put in if it's not in the slides, but this chapter is already getting quite dense for a topic that might be dropped, so I'm hesitant to add it.

Reply:
Maybe this could be added to the more big o chapter?

Reply:
I think the book is a good place toget more into detail, I would prefer having in the book rather than having it in the slides. It is good to understand why we care about reducing from n to log n etc, and understand the limiting factor.

Reply:
@TriHardStudios  I like that idea, if we keep the topic, we can add it to more chapter as it's already relatively short.

Reply:
Ok, I think I'll just add this to the original section